src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -272,6 +275,43 @@ def getDatasetLocations(self, datasets, account=None):
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
         return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        The definition of "location" here is a union of all sites holding at least
+        part of the dataset (defined by the DATASET grouping).
+        :param datasets: list of datasets
+        :param account: rucio account to use, if it is not provided we fallback to
+        pileup account for backward compatibility
+        :return: a dictionary of dataset locations, key'ed by the dataset name
+        """
+        
+        result = {}
+        for dbsUrl, datasets in datasetsWithDbsURL.items():
+            for dataset in datasets:
+                try:
+                    queryDict = {'query': {'pileupName': dataset},
+                                'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                    pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
+                    msPileupUrl = f"https://cmsweb{pileUpinstance}.cern.ch/ms-pileup/data/pileup"
+                    self.logger.debug(f'dbsUrl: {dbsUrl}')
```

Comment:
Perhaps merge these 2 debugging records as well in a single one, e.g.: "Using DbsUrl: {dbsUrl} and MSPileup URL: {msPileupUrl}".
Also move it under the first `for` loop, given that it will not change between datasets of the same url instance.